Core/Inc/bms_can_monitor.h

```diff
@@ -0,0 +1,24 @@
+
+#ifndef BMS_CAN_MONITOR_H
+#define BMS_CAN_MONITOR_H
+
+#include "cmsis_os.h"
+#include "timer.h"
+
+#define CANID_BMS_MONITOR		0x070 /*BMS MONITOR WATCHDOG*/ /*Arbitrary*/
+#define BMS_CAN_MONITOR_DELAY   5000 /*ms*/
+
+typedef struct 
+{
+    nertimer_t timer;
+} bms_can_monitor_t;
```

Comment:
Similar vein to my other comments, make this struct more generic (just `bms` or `shepherd` or something) and then use an RTOS timer which invokes a callback which queues a fault when expired